lib/adapters/http.js

```diff
@@ -126,11 +126,11 @@ module.exports = function httpAdapter(config) {
           });
         }
 
-
         if (shouldProxy) {
           proxy = {
             host: parsedProxyUrl.hostname,
-            port: parsedProxyUrl.port
+            port: parsedProxyUrl.port,
+            protocol: parsedProxyUrl.protocol
```

Comment:
Should the protocol follow the proxy or the request?

Suppose an http request with env,

```
http_proxy=https://proxy.com
```

Well, following the proxy looks fine, because it is the real final endpoint.

Reply:
Yes, this is correct, the proxy protocol should follow the proxy config. The proxy connection should use the protocol specified in the proxy config (e.g., https://proxy.com). Once a connection to the proxy has been established, then original request is sent using the request protocol.

Here's the real world scenario I ran into originally. At work I'm behind a corporate proxy that only supports HTTP.

```
http_proxy=http://corporate.com
https_proxy=http://corporate.com
```

Whenever I would make an HTTPS request through the proxy (e.g., https://www.google.com), Axios would use the request protocol instead of the proxy protocol to connect to the proxy (in this case https://corporate.com) and the connection would fail because corporate.com does not support HTTPS.